File: agent-python/tools/implementations/web_search.py
"""
从网络中搜索相关信息
"""

from typing import Dict, Any, List
import uuid
from ..base import BaseTool, ParameterSchema
from llm.llm_client import llm_call

class WebSearchTool(BaseTool):
    """从网络中搜索相关信息"""

    def get_name(self) -> str:
        return "web_search"
    
    def get_description(self) -> str:
        return "从网络中搜索相关信息。当用户明确需要外部检索时或者你的知识库里明确没有相关信息时候，使用这个工具。"
    
    def get_parameters(self) -> List[ParameterSchema]:
        return [
            ParameterSchema(
                name="search_input",
                description="需要搜索的相关内容。用于去网络查询。",
                type="string",
                required=True
            ),
        ]
    
    def execute(self, parameters: Dict[str, Any]) -> Dict[str, Any]:
        """执行网络搜索"""
        search_input = parameters.get("search_input", "").strip()
        
        if not search_input:
            return {
                "success": False,
                "error": "缺少必要参数：search_input"
            }
        
        try:
            result = llm_call(search_input,model="perplexity/sonar") # 使用perplexity/sonar模型进行网络搜索
            return {
                "success": True,
                "message": result,
                "query": search_input
            }
            
        except Exception as e:
            return {
                "success": False,
                "error": f"网络搜索失败: {str(e)}",
                "query": search_input
            }

File: agent-js/src/tools/implementations/web-search.js
import { createTool } from '../base.js';
import { createParameterSchema } from '../parameter-schema.js';
import { chat } from '../../llm/llm-client.js';

/**
 * 创建网络搜索工具
 */
export function createWebSearchTool() {
  return createTool({
    name: 'web_search',
    description: '通过网络搜索获取信息。当用户提出外部搜索或需要查询最新数据、时事新闻、实时状态或当前知识库无法提供准确答案时使用此工具。',
    parameters: [
      createParameterSchema({
        name: 'search_input',
        description: '具体的搜索查询内容，应包含明确的关键词和上下文信息，以确保搜索结果的准确性和相关性。',
        type: 'string',
        required: true
      })
    ],
    
    async execute(parameters) {
      const { search_input } = parameters;
      
      if (!search_input || !search_input.trim()) {
        throw new Error('缺少必要参数：search_input');
      }

      const searchQuery = search_input.trim();

      try {
        // 使用简化的 chat 接口，会自动获取用户 token
        const result = await chat({
          model: 'perplexity/sonar',
          messages: [{ role: 'user', content: searchQuery }],
          temperature: 0.3,
          maxTokens: 2000
        });

        return {
          message: result,
          query: searchQuery,
          timestamp: new Date().toISOString()
        };

      } catch (error) {
        console.error('Web search failed:', error);
        throw new Error(`网络搜索失败: ${error.message}`);
      }
    },
  });
} 